### PR TITLE
Feature: Add flags to sequences

### PR DESCRIPTION
**Pull request by aadiyatul**
Adds a method `outputForSEQXFileWithFlags` for setting flags for every element forming a sequence.
Useful for auxiliary outputs of Tektronix AWG70000.
Compatible with [Qcodes PR #3450](https://github.com/QCoDeS/Qcodes/pull/3450), mentioned in [discussion #3433](https://github.com/QCoDeS/Qcodes/discussions/3433) in Qcodes.

### CODE REVIEW

File: broadbean/element.py [hunk 1]
```diff
@@ -58,6 +58,26 @@ def addBluePrint(self, channel: Union[str, int],
...
 
+    def addFlags(self, channel: Union[str, int],
+                 flags: List[int]) -> None:
+        """
+        Adds flags for the specified channel.
+        List of 4 flags, each of which should be 0 for 'No change', 1 for 'High',
+        2 for 'Low', 3 for 'Toggle', or 4 for 'Pulse'.
+        """
+        if not isinstance(flags, list):
+            raise ValueError('Flags should be given as a list.')
```
[the remark this answers is not in the document]
Comment (aadiyatul) [reply broadbean/element.py]: I fixed this and added a couple of other minor changes, but the code testing did not start automatically. Could you please launch it?

[the remark this answers is not in the document]
Comment (jenshnielsen) [reply broadbean/element.py]: Thanks, Github does not trigger runs automatically until after your first pr is merge. I have run it

[the remark this answers is not in the document]
Comment (jenshnielsen) [reply broadbean/element.py]: I would recommend investing a bit of time being able to setup and run mypy + tests locally that will give you a much faster feedback loop

[the remark this answers is not in the document]
Comment (aadiyatul) [reply broadbean/element.py]: Thank you. I'd like to ask two more questions:

1. Is it ok to resolve typing errors using type conversion functions? 
E.g. here, the return signature of the method required `List[int]`, whereas the method was returning `nparray[Any]` basing on typing of its internal variables. But we know that in fact the internal variable contains `List[int]`, so I simply used 'ndarray.tolist()' to let mypy pass this. Isn't it considered to be redundant?

2. Could you advise some testing 101 guide, e.g. to start with broadbean tests? Unfortunately I'm really new to it.

[the remark this answers is not in the document]
Comment (jenshnielsen) [reply broadbean/element.py]: 1. Yes that is perfectly fine. 
2. There is not super documentation for this at the moment in broadbean. A good way to figure it out is to actually look at what the CI is doing. https://github.com/QCoDeS/broadbean/blob/master/.github/workflows/pytest.yaml
You should be able to do 

```
pip install -r requirements.txt
pip install .
```
To get all requirements installed and then run either 

```
mypy broadbeawn
```
or 

```
pytest broadbean
```

to run mypy or pytest respectively

[the remark this answers is not in the document]
Comment (aadiyatul) [reply broadbean/element.py]: Thanks for explaining!
Actually on my machine `test_too_short_durations_rejected()` fails for the master branch, without any of my commits. All the other tests pass.

[the remark this answers is not in the document]
Comment (jenshnielsen) [reply broadbean/element.py]: Could it be that you have a too new version of hypothesis installed? If you look at https://github.com/QCoDeS/broadbean/runs/6968014453?check_suite_focus=true you can see that this test starts failing with hypothesis>6.46.7

[the remark this answers is not in the document]
Comment (aadiyatul) [reply broadbean/element.py]: Yes, this was the case, I had the newest one (6.47.3).

**Review by jenshnielsen — APPROVED** [review]
@aadiyatul Thanks, 

@astafan8 I think this is ready. Do you want to have a final look?

File: broadbean/element.py [hunk 2]
```diff
@@ -58,6 +60,45 @@ def addBluePrint(self, channel: Union[str, int],
         self._data[channel] = {}
         self._data[channel]['blueprint'] = newprint
 
+    def addFlags(self, channel: Union[str, int], flags: Sequence) -> None:
```
Comment (astafan8) [inline broadbean/element.py:63]:
```suggestion
    def addFlags(self, channel: Union[str, int], flags: Sequence[Union[str, int]]) -> None:
```

File: broadbean/element.py [hunk 3]
```diff
@@ -58,6 +60,45 @@ def addBluePrint(self, channel: Union[str, int],
...
+                "Flags should be given as a sequence (e.g. a list or a tuple)."
+            )
+
+        if len(flags) != 4:
+            raise ValueError("There should be 4 flags in the list.")
+
+        for cnt, i in enumerate(flags):
+            if i not in [0, 1, 2, 3, 4, "", "H", "L", "T", "P"]:
+                raise ValueError(
+                    'Invalid flag. Allowed flags are 0 or "" (No change), '
```
Comment (astafan8) [inline broadbean/element.py:80]:
```suggestion
                    'Invalid flag at index {cnt}. Allowed flags are 0 or "" (No change), '
```

**Pull request merged**